packages/headless/src/features/folding/insight-folding-actions.ts

```diff
@@ -45,7 +45,7 @@ export const loadCollection = createAsyncThunk<
     {getState, rejectWithValue, extra: {apiClient}}
   ) => {
     const state = getState();
-    const actualRequest = buildInsightLoadCollectionRequest(
+    const actualRequest = await buildInsightLoadCollectionRequest(
```

Comment:
`buildInsightLoadCollectionRequest ` is now an async function, because it uses an async function that augments the request with the analytics section.